libs/vkd3d/swapchain.c

```diff
@@ -1052,23 +1035,56 @@ static VkResult d3d12_swapchain_record_swapchain_blit(struct d3d12_swapchain *sw
         viewport.height = swapchain->vk_swapchain_height;
     }
 
+    image_barrier.sType = VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER;
+    image_barrier.pNext = NULL;
+    image_barrier.srcAccessMask = 0;
+    image_barrier.dstAccessMask = VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
+    image_barrier.oldLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+    image_barrier.newLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+    image_barrier.srcQueueFamilyIndex = VK_QUEUE_FAMILY_IGNORED;
+    image_barrier.dstQueueFamilyIndex = VK_QUEUE_FAMILY_IGNORED;
+    image_barrier.image = swapchain->vk_swapchain_images[dst_index];
+    image_barrier.subresourceRange.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+    image_barrier.subresourceRange.baseMipLevel = 0;
+    image_barrier.subresourceRange.levelCount = 1;
+    image_barrier.subresourceRange.baseArrayLayer = 0;
+    image_barrier.subresourceRange.layerCount = 1;
+
+    if (attachment_info.loadOp != VK_ATTACHMENT_LOAD_OP_DONT_CARE)
+        image_barrier.dstAccessMask = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT;
```

Comment:
Shouldn't this if loadOp == LOAD_OP_LOAD? Also, |= instead of = probably?

Reply:
There was a reason for the DONT_CARE check but I can't quite remember what it was.

missing |= is a typo.